splice_encoding/src/main/java/com/splicemachine/utils/logging/MaskPatternLayout.java

```diff
@@ -0,0 +1,85 @@
+package com.splicemachine.utils.logging;
+
+import org.apache.log4j.Logger;
+import org.apache.log4j.PatternLayout;
+import org.apache.log4j.spi.LoggingEvent;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class MaskPatternLayout extends PatternLayout {
+
+    private String maskString = "***MASKED SENSITIVE INFO***";
+    private Pattern maskPattern = null;
+    private Logger logger = Logger.getLogger(this.getClass());
+
+    public String getMaskPattern() {
+        return maskPattern.toString();
+    }
+
+    public void setMaskPattern(String pattern) {
+        try {
+            maskPattern = Pattern.compile(pattern);
+        } catch (Exception e) {
+            logger.error("Error to compile regex for mask pattern, will not mask anything", e);
+        }
+    }
+
+    public void setMaskString(String maskString) {
+        this.maskString = maskString;
+    }
+
+    public String getMaskString() {
+        return this.maskString;
+    }
+
+    public static String maskMessage(String message, Pattern maskPattern, String maskString) {
+        Matcher matcher = maskPattern.matcher(message);
+
+        if (matcher.find()) {
```

Comment:
To handle the message having multiple substrings matched, we need to change to "while" here.

Reply:
@changli6 Changed the code to make it have the ability to match multiple times